hibernate-core/src/test/java/org/hibernate/test/annotations/formula/FormulaNativeQueryTest.java

```diff
@@ -68,11 +70,42 @@ public void cleanup() {
 	}
 
 	@Test
-	@FailureExpected( jiraKey = "HHH-7525" )
 	public void testNativeQuery() throws Exception {
 		doInHibernate(
 				this::sessionFactory, session -> {
-					Query query = session.createNativeQuery( "SELECT ft.* FROM foo_table ft", Foo.class );
+					Query query = session.createNativeQuery( "SELECT ft.*, abs(locationEnd - locationStart) as distance FROM foo_table ft", Foo.class );
+					List<Foo> list = query.getResultList();
+					assertEquals( 3, list.size() );
+				}
+		);
+	}
+
+	@Test
+	public void testNativeQueryWithAliasProperties() throws Exception {
+		doInHibernate(
+				this::sessionFactory, session -> {
+					NativeQuery query = session.createNativeQuery( "SELECT ft.*, abs(ft.locationEnd - locationStart) as d FROM foo_table ft");
+					query.addRoot( "ft", Foo.class )
+							.addProperty( "id", "id" )
+							.addProperty( "locationStart", "locationStart" )
+							.addProperty( "locationEnd", "locationEnd" )
+							.addProperty( "distance", "d" );
+					List<Foo> list = query.getResultList();
+					assertEquals( 3, list.size() );
+				}
+		);
+	}
+
+	@Test
+	public void testNativeQueryWithAliasSyntax() throws Exception {
+		doInHibernate(
+				this::sessionFactory, session -> {
+					NativeQuery query = session.createNativeQuery(
+							"SELECT ft.id as {ft.id}, ft.locationStart as {ft.locationStart}, ft.locationEnd as {ft.locationEnd}, abs(ft.locationEnd - locationStart) as {ft.distance} FROM foo_table ft")
+							.addEntity( "ft", Foo.class );
+					Map<String,String> props = new HashMap<>();
+					props.put( "distance", "distance" );
+					query.setProperties( props );
```

Comment:
You might well simplify a little bit as
`query.setProperties( Collections.singletonMap( "distance", "distance" ) )`

Reply:
Updated with the recommendation to simplify.